src/hotspot/share/opto/partialEscape.hpp

```diff
@@ -273,6 +294,10 @@ class PEAState {
   void add_new_allocation(GraphKit* kit, Node* obj);
 
   Node* materialize(GraphKit* kit, Node* var);
+  // passive_materialize case-a
```

Comment:
We should not write case-a or case-b. I also do not think that `passive materialization` is the right term we should use in code. It is a made-up term.

I suggest `merge_incoming_virtual` and `merge_incoming_escaped`.

Reply:
good idea.